dashboard/run_data_collection.py

```diff
@@ -0,0 +1,232 @@
+            ]
+        ]
+
+    # Get all lines starting with "Billed" then get the last one as it
+    # shows the total core usage
+    line_interest = [line for line in lines_interest if line.startswith("Billed")][-1]
+
+    try:
+        return int(line_interest.split(" ")[-4].strip().replace(",", ""))
+    except ValueError:
+        print("Failed to convert {} out of \n{}\n to integer.")
+        return None
+
+
+def parse_squeue(lines: Iterable[str]):
+    """Parse the results from the squeue command"""
+    entries = []
+    for ix, line in enumerate(lines):
+        if ix == 0:
+            continue
+        line = line.strip().split()
+
+        try:
+            entries.append(
+                SQueueEntry(
+                    line[0].strip(),
+                    line[1].strip(),
+                    line[4].strip(),
+                    line[3].strip(),
+                    line[7].strip(),
+                    line[8].strip(),
+                    int(line[9].strip()),
+                    int(line[10].strip()),
+                )
+            )
+        except ValueError:
+            print(
+                "Failed to convert squeue line \n{}\n to "
+                "a valid SQueueEntry. Ignored!".format(line)
+            )
+
+    return entries
+
+
+def run_cmd(user: str, hpc: str, cmd: str, timeout: int = 10):
+    """Runs the specified command remotely on the specified hpc using the
+    specified user id.
+
+    Returns False if the command fails for some reason.
+    """
+    ssh_cmd = "ssh {}@{}".format(user, hpc)
+
+    cmd = "{} {}".format(ssh_cmd, cmd)
+    try:
+        result = (
+            subprocess.check_output(cmd, shell=True, timeout=timeout)
+            .decode("utf-8")
+            .strip()
+            .split("\n")
+        )
+    except subprocess.CalledProcessError:
+        print("Cmd {} returned a non-zero exit status.".format(cmd))
+        return False
+    except subprocess.TimeoutExpired:
+        print("The timeout of {} expired for cmd {}.".format(timeout, cmd))
+        return False
+    else:
+        return result
+
+
+def collect_data(user: str, hpc: const.HPC, dashboard_db: str):
+    """Collects data from the specified HPC and adds it to the
+    dashboard db
+    """
+    hpc_value = hpc.value
+
+    # Attempt to open existing database, otherwise create a new one
+    try:
+        dashboard_db = DashboardDB(dashboard_db)
+    except FileNotFoundError:
+        dashboard_db = DashboardDB.create_db(dashboard_db)
+
+    # Get Core hour usage, out of some reason this command is super slow...
+    rt_ch_output = run_cmd(
+        user, hpc.value, "nn_corehour_usage {}".format(PROJECT_ID), timeout=30
+    )
+
+    # Update dashboard db
+    if rt_ch_output:
+        dashboard_db.update_daily_chours_usage(
+            parse_total_chours_usage(rt_ch_output, hpc), hpc
+        )
+    else:
+        return False
+
+    # Squeue
+    sq_output = run_cmd(user, hpc.value, "squeue")
```

Comment:
-A nesi00213

Reply:
I recon log everything for now (have to include the account in the results though) and then add a filter option to the table, which I think dash makes quite easy